tests/test_generic_component.cpp

```diff
@@ -0,0 +1,16 @@
+#define BOOST_TEST_MODULE test module test_generic_component
+#include <boost/test/included/unit_test.hpp>
+
+BOOST_AUTO_TEST_SUITE(generic_suite)
+
+BOOST_AUTO_TEST_CASE(test1) {
```

Comment:
Note to reviewers: these are just dummy tests to verify that our testing works. Actual, meaningful generic component tests will be added in a follow-up PR.

Reply:
I think that as a step before adding meaningful tests that the next step would be to start running these trivial tests in CI. Maybe in this PR?

Reply:
I've got an update to the `fix proto generation` PR in the works that will have to be done before we can get tests running in CI. The short of it is: `boost` doesn't exist in CI by default, and when I added it via `MarkusJx/install-boost`, we still hit problems of CI not knowing where to find it. I'm very happy to defer merging of this PR until that is fixed though, and then make these tests run in CI here :)

Reply:
Either way is fine, I'm happy to defer it to a later PR if there are obstacles to doing it here, and it sounds like there are.

Reply:
Ack, given the existing hurdles I think I'll just merge this when ready, and work on getting CI fixed up in a subsequent PR.